WcaOnRails/app/webpacker/lib/utils/wcif.js

```diff
@@ -86,6 +86,24 @@ export function venueWcifFromRoomId(scheduleWcif, id) {
   return scheduleWcif.venues.find((venue) => venue.rooms.some((room) => room.id === intId));
 }
 
+export function activityWcifFromId(scheduleWcif, id) {
+  return scheduleWcif.venues.flatMap(
+    ({ rooms }) => rooms.flatMap(({ activities }) => activities),
+  ).find((activity) => activity.id === id);
+}
+
+export function doActivitiesMatch(a1, a2) {
+  return a1.activityCode === a2.activityCode
+    && a1.startTime === a2.startTime
+    && a1.endTime === a2.endTime;
```

Comment:
Are you sure this does what it's supposed to?
For example, does the date object representing 01:00AM UTC match the date object representing 02:00AM CET (it should!) and is this intended? Go through the Luxon documentation, there ought to be convenience methods to compare two dates/times in a way that makes it clear whether they need to be the same internal representation (timezone, 12/24hr, etc. all identical) or whether they can be alternative representations (timezones) of the same moment in time

Reply:
Pretty sure, yes. I've tested with FMC Canada and other competitions and it works as expected.
Activity start/end times are stored independent of time zone (always in UTC), so I think you're over-thinking this. Time zones only come into play when displaying the activities to the user.
But let me know if I'm the one under-thinking.

Reply:
According to the documentation of Luxon, comparing the objects themselves via `===` is not a good idea.
See https://moment.github.io/luxon/#/math?id=comparing-datetimes for details

Reply:
These aren't luxon dates, they are strings coming directly from the WCIF.

Reply:
Okay, in that case I understand why `===` makes sense. But the "clean coding" part inside of me wants to make an argument that _especially_ in this case, parsing the dates into Luxon dates and comparing them in the way that is described in the documentation above is required.

Imagine if some third party services send us the UTC dates as `07:30Z` and some send them as `07:30UTC+0`. Both are valid ISO representations as far as I can tell, but I highly doubt we have checks (or even sanitizers) for this in the backend. So even though it might feel a little bit overkill, I'm inclined to say "better safe than sorry".

Reply:
Right, makes sense, will do.
I think [this code](https://github.com/thewca/worldcubeassociation.org/blob/7e1f05453c09410e1baa082cfdb71fa302045cb4/WcaOnRails/app/views/competitions/_competition_schedule_for_venue_table.html.erb#L10) might have the same mistake (and is probably where I got my incorrect assumption from).